internal/fs/tarfs/tarfs_test.go

```diff
@@ -268,3 +268,22 @@ func TestTarFS_Stat_Unsupported(t *testing.T) {
 		}
 	}
 }
+
+func TestTarFS_New_NonTarFile(t *testing.T) {
+	testFiles := []string{
+		"testdata/emptyfile.tar",
+		"testdata/nontar.tar",
+	}
+	for _, name := range testFiles {
+		_, err := New(name)
+		if want := errdef.ErrInvalidTarFile; !errors.Is(err, want) {
+			t.Errorf("TarFS.New(%s) error = %v, wantErr %v", name, err, want)
+		}
+	}
+
+	tarwithoutextension := "testdata/tarwithoutextension"
+	_, err := New(tarwithoutextension)
+	if err != nil {
+		t.Errorf("TarFS.New(%s) error = %v, wantErr %v", tarwithoutextension, err, nil)
```

Comment:
Just realized, should this be put in another test function? As this test function is for non-tar files.

Reply:
Yes this would be better in its own test. I will create a new test `TestTarFS_New_TarWithoutExtension`.